Content/NPCs/Dungeon/CrescentCaster.cs

```diff
@@ -0,0 +1,505 @@
+
+                starColor = new Color(255, 255, 255, 0) * starOpacity;
+                spriteBatch.Draw(starTex, starPos - Main.screenPosition, null, starColor, 0, starTex.Size() / 2, NPC.scale * 0.2f * starSize, SpriteEffects.None, 0f);
+            }
+            return false;
+        }
+
+        public override void OnKill()
+        {
+            ClearBarrierAndBolts(supportTargets);
+
+            if (Main.netMode != NetmodeID.Server)
+            {
+                for (int j = 1; j <= 5; j++)
+                    Gore.NewGoreDirect(NPC.GetSource_Death(), NPC.position + new Vector2(Main.rand.Next(NPC.width), Main.rand.Next(NPC.height)), Main.rand.NextVector2Circular(3, 3), Mod.Find<ModGore>("CrescentCasterGore" + j).Type);
+            }
+        }
+
+        public void DrawPrimitives()
+        {
+            Effect effect = Terraria.Graphics.Effects.Filters.Scene["LightningTrail"].GetShader().Shader;
+
+            Matrix world = Matrix.CreateTranslation(-Main.screenPosition.Vec3());
+            Matrix view = Main.GameViewMatrix.ZoomMatrix;
+            Matrix projection = Matrix.CreateOrthographicOffCenter(0, Main.screenWidth, Main.screenHeight, 0, -1, 1);
+
+            effect.Parameters["time"].SetValue(Main.GameUpdateCount * 0.05f);
+            effect.Parameters["repeats"].SetValue(1f);
+            effect.Parameters["transformMatrix"].SetValue(world * view * projection);
+            effect.Parameters["sampleTexture"].SetValue(ModContent.Request<Texture2D>("StarlightRiver/Assets/GlowTrail").Value);
+            foreach (CrescentCasterBolt bolt in Bolts)
+                bolt.Render(effect);
+        }
+
+        private void CastBehavior()
+        {
+            var tempTargets = ValidTargets(); //purpose of temptargets is to check npcs who were being supported but are no longer
+            var toReduceBarrier = Main.npc.Where(x => x.active && !tempTargets.Contains(x) && supportTargets.Contains(x)).ToList();
+
+            ClearBarrierAndBolts(toReduceBarrier);
+
+            var toAddBolts = Main.npc.Where(x => x.active && tempTargets.Contains(x) && !supportTargets.Contains(x)).ToList();
+
+            foreach(NPC boltNPC in toAddBolts)
+                CreateBolt(boltNPC);
+
+            supportTargets = tempTargets;
+
+            UpdateBolts();
+
+            if (supportTargets.Count == 0)
+            {
+                aiCounter = 390;
+                WalkBehavior();
+                return;
+            }
+
+            switch (xFrame)
+            {
+                case 0: //first frame of casting
+                    xFrame = 1;
+                    NPC.frameCounter = 0;
+                    yFrame = 0;
+                    break;
+                case 1: //Winding up to cast
+                    NPC.frameCounter++;
+
+                    if (NPC.frameCounter % 4 == 0)
```

Comment:
single line conditionals should be inlined

Reply:
fixed